pkg/plugins/repo/client.go

```diff
@@ -145,8 +151,13 @@ func (c *Client) downloadFile(tmpFile *os.File, pluginURL, checksum string, comp
 
 	// Using no timeout as some plugin archives make take longer to fetch due to size, network performance, etc.
 	// Note: This is also used as part of the grafana plugin install CLI operation
-	bodyReader, err := c.sendReqNoTimeout(u, compatOpts)
+	bodyReader, err := c.sendReqNoTimeout(ctx, u, compatOpts)
 	if err != nil {
+		if c.retryCount < 3 {
+			c.retryCount++
+			c.log.Debug("Failed downloading. Will retry once.")
```

Comment:
I guess this log line (same as the original above) would be more correct to just say:
```suggestion
			c.log.Debug("Failed downloading. Will retry.")
```

Reply:
it will retry once... twice :D